File: BackEnd_Learn/Java/todolist/src/main/java/com/leeseojune/todolist/service/TodoService.java
package com.leeseojune.todolist.service;

import com.leeseojune.todolist.dto.TodoDTO;
import com.leeseojune.todolist.entity.Todo;

import java.util.List;

public interface TodoService {
    Iterable<Todo> getAll();
    void addTodo(TodoDTO.Add request);
    void check(int todoId);
    void delete(int todoId);
    void modify(TodoDTO.Modify request, int todoId);
}

File: BackEnd_Learn/NestJS/todolist/src/todo/todo.service.ts
import { Injectable } from '@nestjs/common';
import { resolve } from 'node:path';
import { todoDTO } from './dto/todo.dto';
import { Todo } from './entity/todo.entity';
import { TodoRepository } from './entity/todo.repository';

@Injectable()
export class TodoService {
    constructor(private readonly todoRepository: TodoRepository){}

    getAll(): Promise<Todo[]> {
        return this.todoRepository.find();
    }

    add(body: todoDTO){
        this.todoRepository.save({content: body.content, isCheck: false});
    }

    async check(id: number){
        let todo: Todo = await this.todoRepository.findOne({id: id});
        this.todoRepository.save({id: todo.id, content: todo.content, isCheck: !todo.isCheck});
    }

    async change(id: number, content: string){
        let todo: Todo = await this.todoRepository.findOne({id: id});
        this.todoRepository.save({id: id, content: content, isCheck: todo.isCheck});
    }

    delete(id: number){
        this.todoRepository.delete({id: id});
    }



}
